packages/ssz/src/type/byteVector.ts

```diff
@@ -0,0 +1,84 @@
+import {getNodesAtDepth, Node, packedNodeRootsToBytes, packedRootsBytesToNode} from "@chainsafe/persistent-merkle-tree";
+import {maxChunksToDepth} from "../util/merkleize";
+import {ByteViews} from "./composite";
+import {ByteArrayType} from "./byteArray";
+
+export type ByteVector = Uint8Array;
+
+/* eslint-disable @typescript-eslint/member-ordering */
+
+/**
+ * ByteVector: Immutable alias of Vector[byte, N]
+ * - Notation: `ByteVector[N]`
+ * - Value: `Uint8Array`
+ * - View: `Uint8Array`
+ * - ViewDU: `Uint8Array`
+ *
+ * ByteVector is an immutable value which is represented by a Uint8Array for memory efficiency and performance.
+ * Note: Consumers of this type MUST never mutate the `Uint8Array` representation of a ByteVector.
+ */
+export class ByteVectorType extends ByteArrayType {
+  readonly typeName: string;
+  // Immutable characteristics
+  readonly depth: number;
+  readonly chunkDepth: number;
```

Comment:
should have a `chunkCount` property too

Reply:
Why? As far as I know it's not used by anyone